llvm_objects.py

```diff
@@ -162,7 +205,7 @@ def _get_instructions(self, l_block, w_module):
             prev = current_instr
             w_instrs.append(current_instr)
             l_instr = llwrap.LLVMGetNextInstruction(l_instr)
-        return w_instrs
+        return w_instrs[:]
```

Comment:
Maybe it's clearer if I put it this way:
A is an immutable field
B is a list I'm creating by appending values (it gets changed)
`A = B` would lead to an error: A and B point to the same object which is why RPython thinks you are mutating list A.
Please correct me if I am wrong or if I misunderstood your question.